piptools/scripts/compile.py

```diff
@@ -477,26 +527,43 @@ def cli(
         elif is_setup_file:
             setup_file_found = True
             try:
+                src_dir = os.path.dirname(os.path.abspath(src_file))
```

Comment:
Move this before `try:`. Only one instruction should be within the try-block.

Reply:
Good point.